pybind11-2.7.1-fix-windows-debug.patch

```diff
@@ -0,0 +1,50 @@
+--- a/tools/pybind11Tools.cmake	2022-01-05 20:09:58.470396082 +0000
++++ b/tools/pybind11Tools.cmake	2022-01-05 20:10:09.967351988 +0000
+@@ -113,7 +113,7 @@ if(PYTHON_IS_DEBUG)
+   set_property(
+     TARGET pybind11::pybind11
+     APPEND
+-    PROPERTY INTERFACE_COMPILE_DEFINITIONS Py_DEBUG)
++    PROPERTY INTERFACE_COMPILE_DEFINITIONS PYBIND11_DEBUG)
+ endif()
+ 
+ set_property(
+--- a/tools/pybind11NewTools.cmake	2022-01-05 20:10:04.286373776 +0000
++++ b/tools/pybind11NewTools.cmake	2022-01-05 20:10:30.186274440 +0000
+@@ -123,7 +123,7 @@ if(PYTHON_IS_DEBUG)
+   set_property(
+     TARGET pybind11::pybind11
+     APPEND
+-    PROPERTY INTERFACE_COMPILE_DEFINITIONS Py_DEBUG)
++    PROPERTY INTERFACE_COMPILE_DEFINITIONS PYBIND11_DEBUG)
+ endif()
+ 
+ # Check on every access - since Python2 and Python3 could have been used - do nothing in that case.
+--- a/include/pybind11/detail/common.h	2022-01-05 20:08:54.029643196 +0000
++++ b/include/pybind11/detail/common.h	2022-01-05 20:09:12.178573603 +0000
+@@ -137,7 +137,7 @@
+ #  pragma warning(push)
+ // C4505: 'PySlice_GetIndicesEx': unreferenced local function has been removed (PyPy only)
+ #  pragma warning(disable: 4505)
+-#  if defined(_DEBUG) && !defined(Py_DEBUG)
++#  if defined(_DEBUG) && !defined(PYBIND11_DEBUG)
+ #    define PYBIND11_DEBUG_MARKER
+ #    undef _DEBUG
+ #  endif
+--- a/CMakeLists.txt	2022-01-06 14:02:51.005506594 +0000
++++ b/CMakeLists.txt	2022-01-06 14:08:09.755079249 +0000
+@@ -16,6 +16,14 @@ else()
+   cmake_policy(VERSION 3.18)
+ endif()
+ 
++# In CMake version 3.22 or above, cmake_dependent_option now supports Full Condition Syntax
++# (see https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html).
```

Comment:
Where does this warning happen, and what does it look like?

Reply:
> Where does this warning happen, and what does it look like?

It happens when "building" the `pybind11_vendor` package, at https://github.com/pybind/pybind11/blob/v2.7.1/CMakeLists.txt#L98-L99.  The warning looks like this:

```
Policy CMP0127 is not set: cmake_dependent_option() supports full Condition
Syntax.  Run "cmake --help-policy CMP0127" for policy details.  Use the
cmake_policy command to set the policy and suppress this warning.
```

From my reading of https://cmake.org/cmake/help/latest/policy/CMP0127.html , I don't think we would need *any* changes to this particular line of code, but that doesn't seem to be the case.  No amount of quoting I did on that line made the warning go away, so I just gave up and set CMP0127 to "OLD".  If you have another idea on how to fix it, I'm happy to try that instead.